cpp/dolfinx/la/MatrixCSR.h

```diff
@@ -224,40 +246,40 @@ class MatrixCSR
 
   /// Set values in the matrix
   /// @note Only entries included in the sparsity pattern used to
-  ///       initialize the matrix can be set
-  /// @note All indices are local to the calling MPI rank
-  ///       and entries cannot be set in ghost rows.
-  /// @note This should be called after `finalize`. Using before `finalize` will
-  ///       set the values correctly, but incoming values may get added to them
-  ///       during a subsequent finalize operation.
+  /// initialize the matrix can be set
+  /// @note All indices are local to the calling MPI rank and entries
+  /// cannot be set in ghost rows.
+  /// @note This should be called after `finalize`. Using before
+  /// `finalize` will set the values correctly, but incoming values may
+  /// get added to them during a subsequent finalize operation.
   /// @param[in] x The `m` by `n` dense block of values (row-major) to
-  ///       set in the matrix
+  /// set in the matrix
   /// @param[in] rows The row indices of `x`
   /// @param[in] cols The column indices of `x`
   void set(const xtl::span<const T>& x,
            const xtl::span<const std::int32_t>& rows,
            const xtl::span<const std::int32_t>& cols)
   {
     assert(x.size() == rows.size() * cols.size());
-    const std::int32_t max_row = _index_maps[0]->size_local() - 1;
     for (std::size_t r = 0; r < rows.size(); ++r)
     {
       // Columns indices for row
       std::int32_t row = rows[r];
-      if (row > max_row)
-        throw std::runtime_error("Local row out of range");
 
       // Current data row
       const T* xr = x.data() + r * cols.size();
 
+#ifndef NDEBUG
+      if (!(row + 1 < (int)_row_ptr.size()))
```

Comment:
This is not quite right, I think. When calling set, it should not be allowed to set in the ghost rows, but _row_ptr contains the ghost rows too.